lua/nordic/tests/options.lua

```diff
@@ -1,55 +1,168 @@
--- First test the default config, and then test different variations of the config.
-
-local config = require('nordic.config').options
+local assert_eq = require('nordic.tests').assert_eq
 local load = require('nordic').load
+local U = require('nordic.utils')
+
+local function highlight(group)
+  local function hexify(value)
+    if type(value) == 'number' then
+      return string.format('#%X', value)
+    elseif type(value) == 'table' then
+      return vim.tbl_map(hexify, value)
+    end
+    return value
+  end
 
-local function flip_string(string)
-    local switch = {
-        ['light'] = 'dark',
-        ['dark'] = 'light',
-        ['flat'] = 'classic',
-        ['classic'] = 'flat',
-    }
-    return switch[string]
+  return hexify(vim.api.nvim_get_hl(0, { name = group, create = false }))
 end
 
-load(config)
+load({})
 
-config.on_palette = function(palette)
-    palette.black0 = '#000000'
-end
+-- tests for changes in palette should check highlights (to make sure everything is applied)
 
-config.after_palette = function(palette)
-    local U = require('nordic.utils')
-    palette.bg_visual = U.blend(palette.orange.base, palette.bg, 0.15)
-end
+-- on_palette
+assert_eq(highlight('Normal').bg, '#242933', 'on_palette: all highlights that use `gray0` should be `#242933` by default')
+load({ on_palette = function(palette) palette.gray0 = '#FFFFFF' end })
+assert_eq(highlight('Normal').bg, '#FFFFFF', 'on_platte: changing a color should cascade to all highlights that use it')
+load({ on_palette = function(_palette) end })
+assert_eq(highlight('Normal').bg, '#242933', 'on_palette: reloading should revert the palette to its original state')
 
-config.on_highlight = function(highlights, palette)
-    highlights.TelescopePromptTitle = {
-        fg = palette.red.bright,
-        bg = palette.green.base,
-        italic = true,
-        underline = true,
-        sp = palette.yellow.dim,
-        undercurl = false,
-    }
-end
+-- after_palette
+assert_eq(highlight('Normal').bg, '#242933', 'after_palette: all highlights that use `bg` should be `#242933` by default')
+load({ after_palette = function(palette) palette.bg = '#FFFFFF' end })
+assert_eq(highlight('Normal').bg, '#FFFFFF',
+  'after_platte: changing a color should cascade to all highlights that use it')
+load({ after_palette = function(_palette) end })
+assert_eq(highlight('Normal').bg, '#242933', 'after_palette: reloading should revert the palette to its original state')
+
+-- on_highlight
+assert_eq(highlight('Normal').bg, '#242933', 'on_highlight: `Normal` should be `#242933` by default')
+load({ on_highlight = function(highlights, _palette) highlights['Normal'].bg = '#FFFFFF' end })
+assert_eq(highlight('Normal').bg, '#FFFFFF', 'on_highlight: changing a highlight should actually change the highlight')
+load({ on_highlight = function(_highlights, _palette) end })
```

Comment:
Nit: In lua, use `_` for unused variables.  You can use it multiple times.

Reply:
I don't like that because then you can't tell what that args name was... but it does get the lsp to stop complaining.

[fix(tests): rename unused variables;](https://github.com/AlexvZyl/nordic.nvim/pull/150/commits/389c753cd538b1a1ae09ff4da92cfa21894ddce8)

Reply:
Yeah I agree, but the LSP bothers me more :joy: